src/code/views/node-view.coffee

```diff
@@ -1,8 +1,70 @@
-{div, i, img} = React.DOM
+{input, div, i, img} = React.DOM
+tr = require "../utils/translate"
+
+NodeTitle = React.createFactory React.createClass
+  displayName: "NodeTitle"
+
+  getDefaultProps: ->
+    defaultValue: tr "~NODE.UNTITLED"
+
+
+  componentWillUnmount: ->
+    if @props.isEditing
+      @inputElm().off()
+
+  componentDidUpdate: ->
+    if @props.isEditing
+      $elem =@inputElm()
+      $elem.focus()
+
+      $elem.off()
+      enterKey = 13
+      $elem.on "keyup", (e)=>
+        if e.which is enterKey
+          @finishEditing()
+
+  inputElm: ->
+    $(@refs.input.getDOMNode())
+
+  inputValue: ->
+    @inputElm().val()
+
+  updateTitle: (e) ->
+    newTitle = @inputValue()
+    newTitle = if newTitle.length > 0 then newTitle else @props.defaultValue
```

Comment:
CoffeeScript has one nice conditional assignment sugar you could use here:

    newTitle ||= @props.defaultValue

that evaluates to
   
   newTitle || (newTitle = this.props.defaultValue);